### PR TITLE
Update docs for "module" to "app" changes

### LINKED ISSUE #630: Docs should reflect the terminology change from module to app
As a developer, I'd like to update all the docs to change any `module` reference with `app`. 

**Acceptance:**
- all the [relevant sections](https://github.com/spring-cloud/spring-cloud-dataflow/tree/master/spring-cloud-dataflow-docs/src/main/asciidoc) are updated from `module` to `app` [_ex: `module register` to `app register`_]

### PR DESCRIPTION
**Pull request by markfisher**
resolves #630

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc [hunk 1]
```diff
@@ -45,10 +45,10 @@ image::dataflow-available-apps-list.png[List of available applications, scaledwi
...
 The _Runtime_ section of the Dashboard application shows the Spring Cloud Data Flow
-cluster view with the list of all running applications. For each runtime module the
+cluster view with the list of all running applications. For each runtime app the
 state of the deployment and the number of deployed instances is shown.
 A list of the used deployment properties is available by clicking on the
-module id.
+app id.
 
 .List of Running Applications
 image::dataflow-runtime.png[List of running applications, scaledwidth="100%"]
```
Comment (cppwfs) [inline spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc]: Image still shows module

File: spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc [hunk 2]
```diff
@@ -68,45 +68,45 @@ image::dataflow-streams-list-definitions.png[List of Stream Definitions, scaledw
...
-[[dashboard-tasks-modules]]
-=== Modules
+[[dashboard-tasks-apps]]
+=== Apps
 
-_Modules_ encapsulate a unit of work into a reusable component. Within the Dataflow
-runtime environment Modules allow users to create definitions for _Streams_ as
-well as _Tasks_. Consequently, the _Modules_ tab within the _Tasks_ section
+_Apps_ encapsulate a unit of work into a reusable component. Within the Dataflow
+runtime environment Apps allow users to create definitions for _Streams_ as
```
Comment (cppwfs) [inline spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc]: Do we want to italicize _Apps_

File: spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc [hunk 3]
```diff
@@ -68,45 +68,45 @@ image::dataflow-streams-list-definitions.png[List of Stream Definitions, scaledw
...
 
-[[dashboard-tasks-modules]]
-=== Modules
+[[dashboard-tasks-apps]]
+=== Apps
 
-_Modules_ encapsulate a unit of work into a reusable component. Within the Dataflow
-runtime environment Modules allow users to create definitions for _Streams_ as
-well as _Tasks_. Consequently, the _Modules_ tab within the _Tasks_ section
+_Apps_ encapsulate a unit of work into a reusable component. Within the Dataflow
```
Comment (cppwfs) [inline spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc]: Do we want to refer to Dataflow as `Data Flow`

Comment (markfisher) [inline spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc]: yes! I thought I caught them all

File: spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc (hunk 2 above)
Comment (markfisher) [inline spring-cloud-dataflow-docs/src/main/asciidoc/dashboard.adoc]: I guess so, although in hindsight _this_ section _seems_ to _overuse_ italics _possibly_

**Comment by cppwfs** [comment]
Images for App lists still shows modules in the URI:    https://github.com/spring-cloud/spring-cloud-dataflow/pull/636/files#diff-a98c4a2bb1041515f57726a49048e1d5R64

**Comment by cppwfs** [comment]
LGTM will clean up the comments above on merge.

**Comment by cppwfs** [comment]
Merged, Changes applied during merge.

**Pull request closed without merging**